lib/connections/service.go

```diff
@@ -370,15 +382,16 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
```

Comment:
More generally, I think we should make up our mind about how we handle configs. I think that in things like initialization and startup it's fine to be explicit and pass the configs in parameters - this is also necessary when the code is intended to be run from a CommitConfiguration step which is similar to during startup.

In the other case where we have something long running that checks for an option one a minute or so using the global wrapper is appropriate - it doesn't need to react to the latest change, it just needs to be pick up the new value at some point...